src/completion_queue.rs

```diff
@@ -59,22 +78,20 @@ impl CompletionQueue {
         Ok(())
     }
 
-    /// Poll `num_entries` work completions from CQ
-    pub(crate) fn poll(&self, num_entries: usize) -> io::Result<Vec<WorkCompletion>> {
-        let mut ans: Vec<WorkCompletion> = Vec::with_capacity(num_entries);
-        let poll_res =
-            unsafe { ibv_poll_cq(self.as_ptr(), num_entries.cast(), ans.as_mut_ptr().cast()) };
-        if poll_res >= 0_i32 {
-            assert!(num_entries >= poll_res.cast());
-            let poll_res = poll_res.cast();
-
+    /// Poll work completions from CQ up to `DEFAULT_CQ_SIZE` at a time
+    pub(crate) fn poll_cq_multiple(&self) -> io::Result<Vec<WorkCompletion>> {
+        // let mut wcs_buf = [MaybeUninit::<WorkCompletion>::uninit(); DEFAULT_CQ_SIZE_USIZE];
+        let mut wc_buf: Vec<WorkCompletion> = Vec::with_capacity(self.max_cqe.cast());
```

Comment:
It's better to accept a mutable buffer as a function argument. So the caller can reuse the buffer without extra allocation.

Reply:
Yes, input mutable buffer and output cqe_num is better.